calyx/src/passes/top_down_static_timing.rs

```diff
@@ -324,39 +324,23 @@ impl Schedule<'_> {
             .with_pos(&con.attributes));
         }
 
-        let port_guard: ir::Guard = Rc::clone(&con.port).into();
-
-        let preds = self.calculate_states(
-            &con.body,
-            cur_state,
-            &pre_guard.clone().and(port_guard.clone()),
-        )?;
-
-        let body_exit = preds
-            .iter()
-            .max_by_key(|(state, _)| state)
-            .unwrap_or(&(cur_state, pre_guard.clone()))
-            .0
-            + 1;
+        let mut body_exit = cur_state;
+
+        for _ in 0..*con.attributes.get("bound").unwrap() {
```

Comment:
The unwrap will error out if `bound` is missing. Does the calling code ensure that `@bound` is present?

Reply:
It is checking for @static, but I will double check @bound is there too